packages/wrangler/src/triggers/deploy.ts

```diff
@@ -0,0 +1,256 @@
+import chalk from "chalk";
+import { fetchListResult, fetchResult } from "../cfetch";
+import {
+	ensureQueuesExist,
+	formatTime,
+	publishCustomDomains,
+	publishRoutes,
+	renderRoute,
+	sleep,
+	updateQueueConsumers,
+} from "../deploy/deploy";
+import { UserError } from "../errors";
+import { logger } from "../logger";
+import { getWorkersDevSubdomain } from "../routes";
+import { getZoneForRoute } from "../zones";
+import type { Config } from "../config";
+import type { Route } from "../config/environment";
+import type { RouteObject } from "../deploy/deploy";
+
+type Props = {
+	config: Config;
+	accountId: string | undefined;
+	name: string | undefined;
+	env: string | undefined;
+	triggers: string[] | undefined;
+	routes: string[] | undefined;
+	legacyEnv: boolean | undefined;
+	dryRun: boolean | undefined;
+	dispatchNamespace: string | undefined;
+	experimentalVersions: boolean | undefined;
+};
+
+export default async function triggersDeploy(props: Props): Promise<void> {
+	const { config, accountId, name: scriptName } = props;
+
+	const triggers = props.triggers || config.triggers?.crons;
+	const routes =
+		props.routes ?? config.routes ?? (config.route ? [config.route] : []) ?? [];
+	const routesOnly: Array<Route> = [];
+	const customDomainsOnly: Array<RouteObject> = [];
+	for (const route of routes) {
+		if (typeof route !== "string" && route.custom_domain) {
+			if (route.pattern.includes("*")) {
+				throw new UserError(
+					`Cannot use "${route.pattern}" as a Custom Domain; wildcard operators (*) are not allowed`
+				);
+			}
+			if (route.pattern.includes("/")) {
+				throw new UserError(
+					`Cannot use "${route.pattern}" as a Custom Domain; paths are not allowed`
+				);
+			}
+			customDomainsOnly.push(route);
+		} else {
+			routesOnly.push(route);
+		}
+	}
```

Comment:
You've added logic for publishing routes here, but haven't removed it from `wrangler deploy`. Are you sure this won't cause duplicate deploys?

Reply:
Definitely not causing duplicate deploys. This was left for the validation. The arrays are dead code which have been removed in https://github.com/cloudflare/workers-sdk/pull/5436